rust/connlib/libs/common/src/control.rs

```diff
@@ -342,7 +363,11 @@ impl PhoenixSenderWithTopic {
     /// Sends a message to the associated topic using a [PhoenixSender] also setting the ref
     ///
     /// See [PhoenixSender::send]
-    pub async fn send_with_ref(&mut self, payload: impl Serialize, reference: i32) -> Result<()> {
+    pub async fn send_with_ref(
+        &mut self,
+        payload: impl Serialize,
+        reference: impl Into<String>,
```

Comment:
Using `impl ToString` instead of `impl Into<String>` is more flexible, depending on what we want to do with this; I noticed that we're calling `to_string` manually on two of our `send_with_ref` calls, so it seems appropriate.